src/SurveyService.Business/Commands/Question/GetQuestionCommand.cs

```diff
@@ -0,0 +1,114 @@
+  private readonly IHttpContextAccessor _httpContextAccessor;
+
+  public GetQuestionCommand(
+    IHttpContextAccessor httpContextAccessor,
+    IQuestionRepository questionRepository, 
+    IQuestionResponseMapper questionResponseMapper,
+    IResponseCreator responseCreator,
+    IUserService userService,
+    IOptionInfoMapper optionInfoMapper,
+    IUserAnswerInfoMapper userAnswerInfoMapper)
+  {
+    _httpContextAccessor = httpContextAccessor;
+    _repository = questionRepository;
+    _questionResponseMapper = questionResponseMapper;
+    _responseCreator = responseCreator;
+    _userService = userService;
+    _optionInfoMapper = optionInfoMapper;
+    _userAnswerInfoMapper = userAnswerInfoMapper;
+  }
+  
+  public async Task<OperationResultResponse<QuestionResponse>> ExecuteAsync(GetQuestionFilter filter)
+  {
+
+    OperationResultResponse<QuestionResponse> response = new();
+    if (filter is null)
+    {
+      return _responseCreator.CreateFailureResponse<QuestionResponse>(
+        HttpStatusCode.BadRequest,
+        new List<string> { "You must enter 'questionid'." });
+    }
+
+    DbQuestion dbQuestion = await _repository.GetAsync(filter);
+    
+    if (dbQuestion.Deadline > DateTime.Now && !dbQuestion.HasRealTimeResult)
+    {
+      return _responseCreator.CreateFailureResponse<QuestionResponse>(
+        HttpStatusCode.Forbidden,
+        new List<string> { "Question has no real-time results, result can be reached only after deadline." });
+    }
+
+    if (dbQuestion.IsPrivate && _httpContextAccessor.HttpContext.GetUserId() != dbQuestion.CreatedBy)
+    {
+      return _responseCreator.CreateFailureResponse<QuestionResponse>(
+        HttpStatusCode.Forbidden,
+        new List<string> { "Question is private, only author can view the results." });
+    }
+    
+    if (dbQuestion.IsAnonymous && filter.IncludeUserInfo)
+    {
+      filter.IncludeUserInfo = false;
+      response.Errors.Add("Personal information about user can't be loaded as question is anonymous.");
+    }
+    
+    List<OptionInfo> optionsInfo = new();
+
+    if (filter.IncludeOptions)
+    {
+      foreach (DbOption option in dbQuestion.Options)
+      {
+        List<UserAnswerInfo> userAnswersInfo = new();
+
+        foreach (DbUserAnswer optionUsersAnswer in option.UsersAnswers)
+        {
+          UserData userData = filter.IncludeUserInfo
+            ? (await _userService.GetUsersDataAsync(new List<Guid> { optionUsersAnswer.UserId }, null)).FirstOrDefault()
+            : null;
+          UserAnswerInfo userAnswerInfo = _userAnswerInfoMapper.Map(optionUsersAnswer, userData);
+          userAnswersInfo.Add(userAnswerInfo);
+        }
+
+        OptionInfo optionInfo = _optionInfoMapper.Map(option, userAnswersInfo);
+        optionsInfo.Add(optionInfo);
+      }
+
+      response.Body = _questionResponseMapper.Map(dbQuestion, optionsInfo);
+    }
+    else
+    {
+      response.Body = _questionResponseMapper.Map(dbQuestion);
```

Comment:
use in command only questionResponseMapper; call optionInfoMapper in questionResponseMapper, userAnswerInfoMapper in questionResponseMapper and so on

Reply:
Well, is what I did right?

Reply:
Sorry, my bad. It's all-right about mapping in common. But what for you create OptionInfo, UserInfo.. in command? You need just dbQuestion and usersData

Reply:
Done in waterfall way